### PR TITLE
fix: fix tokenRatesController

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

When Using latest version of core, the UI cannot display on the home page the user's fiat token balance.
This PR fixes the instantiation of the tokenRatesController.

This was tested by building core with assets-controllers version 21.0.0.

## **Related issues**

Fixes: 

## **Manual testing steps**

1. Build core
2. build extension
3. go to home page
4. Click import token button
5. Choose any token and import
6. You should see on the home page the token balance even if $0.00

### **Before**

With the current versions you would have errors loading the extension 
image

### **After**

Loads successfully and home page displays token fiat balance
image

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by jiexi** [comment]
https://github.com/MetaMask/metamask-extension/pull/22152 should have included this change. Thank you for fixing this

**Review by jiexi — CHANGES_REQUESTED** [review]
GasFeeController also takes in `onNetworkDidChange` now

**Comment by bergeron** [comment]
@metamaskbot update-policies

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -698,7 +702,7 @@ export default class MetamaskController extends EventEmitter {
         this.networkController.getProviderAndBlockTracker().provider,
       // NOTE: This option is inaccurately named; it should be called
       // onNetworkDidChange
```
Comment (jiexi) [inline app/scripts/metamask-controller.js:704]:
```suggestion
```

probably don't need this comment anymore

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -511,7 +513,7 @@ export default class MetamaskController extends EventEmitter {
         // that is what the handler name implies, but this triggers too soon
         // causing the provider of the AssetsContractController to trail the
         // network provider by one update.
```
Comment (jiexi) [inline app/scripts/metamask-controller.js:515]: this comment block is now stale and should probably be removed

**Review by jiexi — COMMENTED** [review]
Looking good. Just minor comments on removing some stale comments. Thank you!

File: app/scripts/metamask-controller.js (hunk 2 above)
Comment (sahar-fehri) [reply app/scripts/metamask-controller.js:515]: 🙏

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (sahar-fehri) [reply app/scripts/metamask-controller.js:704]: 🙏

**Review by jiexi — APPROVED** [review]

**Comment by jiexi** [comment]
REMINDER: Please update the squashed commit message to mention the bump of `@metamask/assets-controller` to `21.0.0`

**Review by mcmire — APPROVED** [review]
Looks great! Thank you for taking care of this.

**Pull request merged**